src/server/pkg/sql/sql.go

```diff
@@ -0,0 +1,80 @@
+package sql
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+)
+
+// PGDumpReader parses a pgdump file into a header, rows, and a footer
+type PGDumpReader struct {
+	Header []byte
+	Footer []byte
+	rd     *bufio.Reader
+}
+
+// NewPGDumpReader creates a new PGDumpReader
+func NewPGDumpReader(r *bufio.Reader) *PGDumpReader {
+	return &PGDumpReader{
+		rd: r,
+	}
+}
+
+// ReadRows parses the pgdump file and populates the header and the footer
+// It returns EOF when done, and at that time both the Header and Footer will
+// be populated. Both header and footer are required. If either are missing, an
+// error is returned
+func (r *PGDumpReader) ReadRows(count int64) (rowsDump []byte, rowsRead int64, err error) {
```

Comment:
I'm generally not a fan of using named return parameters, except for causes where you need reference them in closures (ie `retErr`) I've shot myself in the foot a bunch of times by inadvertently shadowing the parameters. I also don't think these are really making the code much shorter, they're just saving you writing a few `:=`s instead of `=`s I think. You also probably won't need `_err` if you don't use up that name in the return parameter.

Reply:
Makes sense. Fixed.